### PR TITLE
Add configure GPG locations

### PR DESCRIPTION
**Pull request by mamedin**
When the `archivematica_src_configure_gpg` is defined, this commit adds
a new GPG space if it doesn't already exist and configures a GPG AIPs
Store and a GPG Transfer Backlog.

The main purpose of the automatic configuration of GPG space and
locations is for QA and AMAUAT. For production sites it is better to
create GPG Space and locations from the GUI.

Connects to https://github.com/archivematica/Issues/issues/832

### CODE REVIEW

File: tasks/configure-gpg.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+- name: "Install jq package"
+  package: 
+    name: "jq"
+  become: yes
+
+# check when GPG Space already exists
+
+- name: "Count GPG Spaces"
+  shell: > 
+    curl -s -X GET
```
Comment (scollazo) [inline tasks/configure-gpg.yml]: I would prefer to use always the url ansible module, and set the fact in a different task, using jq or jinja2 filters.

File: tasks/configure-gpg.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    warn: no
+    
+# add GPG space and locations when there is no GPG space already configured
+- name: "Add GPG space and locations"
+  block:
+
+    - name: "Set gpg binary name"
+      set_fact:
+        gpg_binary: "gpg1"
+      when: "ansible_distribution_version | version_compare('18.04', '>=')"
```
Comment (scollazo) [inline tasks/configure-gpg.yml:21]: Is this correct? We don't support ubuntu >= 18.04

File: tasks/configure-gpg.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    - name: "Create batch file for GPG key"
+      template: 
+        src: "gpg/gpg-batch-file.j2"
+        dest: "/tmp/gpg-batch-file"
+      become: "yes"
+
+    - name: "Check whether GPG already exists"
+      shell: >
+        {{ archivematica_src_configure_gpg.binary }} --no-tty
```
Comment (scollazo) [inline tasks/configure-gpg.yml]: Could you split each option into it's own line?

File: templates/gpg/gpg-batch-file.j2 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+%echo Generating a basic OpenPGP key
+Key-Type: RSA
+Key-Length: 4096
+Subkey-Type: ELG-E
+Subkey-Length: 4096
+Name-Real: {{ archivematica_src_configure_gpg.key_user }}
+Name-Comment: {{ archivematica_src_configure_gpg.key_comment }}
+Name-Email: {{ archivematica_src_configure_gpg.key_mail }}
+Expire-Date: 0
+Passphrase: passphrase
```
Comment (scollazo) [inline templates/gpg/gpg-batch-file.j2]: Can we use a configuration parameter for this, and if it's not available just use a blank passphrase?

**Review by scollazo — CHANGES_REQUESTED** [review]

File: tasks/configure-gpg.yml (hunk 2 above)
Comment (mamedin) [reply tasks/configure-gpg.yml:21]: Yes, for future versions :P

File: templates/gpg/gpg-batch-file.j2 (hunk 4 above)
Comment (mamedin) [reply templates/gpg/gpg-batch-file.j2]: We are using empty string when creating the keys from GUI:

https://github.com/artefactual/archivematica-storage-service/blob/stable/0.14.x/storage_service/common/gpgutils.py#L31

We can add a new field to the dict for the passphrase (it's free), good idea. Thanks!

**Comment by mamedin** [comment]
Hi Santi:

* removed the jq/curl dependency: Using uri module and json_query filter
* Added passphrase var
* changed shell tasks with 1 line for each option.

File: tasks/configure-gpg.yml (hunk 2 above)
Comment (scollazo) [reply tasks/configure-gpg.yml:21]: It's always good to think about the future ;)

File: tasks/configure-gpg.yml [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    - name: "Set gpg binary name"
+      set_fact:
+        gpg_binary: "gpg1"
+      when: "ansible_distribution_version | version_compare('18.04', '>=')"
+
+    - name: "Create batch file for GPG key"
+      template:
+        src: "gpg/gpg-batch-file.j2"
+        dest: "/tmp/gpg-batch-file"
```
Comment (scollazo) [inline tasks/configure-gpg.yml:26]: Is this file deleted after the process finishes?

**Review by scollazo — APPROVED** [review]
Other than removing the temporary gpg batch file, this is ready for merge!

File: tasks/configure-gpg.yml (hunk 5 above)
Comment (mamedin) [reply tasks/configure-gpg.yml:26]: Good catch!!

**Pull request merged**